docs/developer/watcher-local-test.md

```diff
@@ -0,0 +1,136 @@
+data:
+  config: $(echo "${$(k3d kubeconfig get skr-local)/0.0.0.0/host.k3d.internal}" | base64)
+---
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: Kyma
+metadata:
+  name: kyma-sample
+  namespace: default
+spec:
+  channel: regular
+  sync:
+    enabled: true
+  modules:
+    - name: kcp-module
+EOF
+```
+### Verify that watcher-webhook is installed on SKR
+By checking the `Kyma CR` events, verify that the `SKRWebhookIsReady` ready condition is set to `True`.
+```yaml
+Status:                                              
+    Conditions:                                        
+        Message:               skrwebhook is synchronized
+        Observed Generation:   1               
+        Reason:                SKRWebhookIsReady
+        Status:                True
+        Type:                  Ready
+```
+### Edit kyma on SKR
+1. Run the following command to switch the context to using the SKR cluster
+```shell
+kubectl config use-context k3d-skr-local
+```
+2. Add the following line to the Kyma CR `spec` to trigger the watcher KCP update
+```yaml
+  modules:
+  - name: skr-module
+```
+### verify lifecycle-manager logs
+By watching the lifecycle-manager logs, verify that the listener is logging messages indicating the reception of a message from the watcher
+```log
+{"level":"INFO","date":"2022-12-16T10:52:07.983431734Z","logger":"request–verifier","caller":"security/san_pinning.go:85","msg":"###### Request Header [By=spiffe://cluster.local/ns/kcp-system/sa/klm-controller-manager;Hash=d28f48b3777e8431cdfce5f36da0e1b1f239de7c9c2ec3313ef163f3918610a0;Subject=\"\";URI=spiffe://cluster.local/ns/istio-system/sa/istio-ingressgateway-service-account]","context":{}} 
+{"level":"INFO","date":"2022-12-16T10:52:07.98353109Z","caller":"event/skr_events_listener.go:103","msg":"request could not be verified - Event will not be dispatched","context":{"Module":"Listener","error":"empty certificate"}}       
+{"level":"INFO","date":"2022-12-16T10:52:13.406499949Z","caller":"controllers/kyma_controller.go:87","msg":"reconciling modules","context":{"controller":"kyma","controllerGroup":"operator.kyma-project.io","controllerKind":"Kyma","kyma":{"name":"kyma-sample","namespace":"default"},"namespace":"default","name":"kyma-sample","reconcileID":"64add714-e3fb-4513-827d-9bb9ba2325a3"}}                                                                                          
+{"level":"INFO","date":"2022-12-16T10:52:13.451737025Z","caller":"controllers/kyma_controller.go:206","msg":"syncing state","context":{"controller":"kyma","controllerGroup":"operator.kyma-project.io","controllerKind":"Kyma","kyma":{"name":"kyma-sample","namespace":"default"},"namespace":"default","name":"kyma-sample","reconcileID":"64add714-e3fb-4513-827d-9bb9ba2325a3","state":"Processing"}}
+```
+### verify SKR-watcher logs
+By watching the `skr-webhook` deployment's logs, verify that the KCP request is sent successfully
```

Comment:
since your guide does not assume creating a deployment for module manager, the modules are never reconciled and stay stuck in processing with the Kyma CR. thus I was not able to verify this step with the guide.